### PR TITLE
BIGTOP-2865: HBase smoke test implementation

### PR DESCRIPTION
**Pull request by JunHe77**
Change-Id: I5731baebc41090903fee8b79d0f4013dbe62c5b7
Signed-off-by: Jun He <jun.he@arm.com>

### CODE REVIEW

File: bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestImportTsv.groovy [hunk 1]
```diff
@@ -89,7 +92,8 @@ public class TestImportTsv {
     // delete data and junk from HDFS
     sh.exec("hadoop fs -rmr -skipTrash $DATADIR1",
       "hadoop fs -rmr -skipTrash $DATADIR2",
-      "hadoop fs -rmr -skipTrash /user/$USER/partitions_*");
+      "hadoop fs -rmr -skipTrash /user/$USER/hbase*",
```
Comment (evans-ye) [inline bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestImportTsv.groovy]: Shall we change this to rm -r at the same time?

File: bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestHBaseCompression.groovy [hunk 2]
```diff
@@ -85,7 +85,7 @@ class TestHBaseCompression {
   }
 
   @Test
-  @Ignore("BIGTOP-218")
+  //@Ignore("BIGTOP-218")
```
Comment (evans-ye) [inline bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestHBaseCompression.groovy]: What's the reason for comment out this?

File: bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestImportTsv.groovy (hunk 1 above)
Comment (JunHe77) [inline bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestImportTsv.groovy]: Got it and updated. I'm not sure about the context of using 'rmr', so I leave it unchanged in initial commit. :P

File: bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestHBaseCompression.groovy (hunk 2 above)
Comment (JunHe77) [inline bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestHBaseCompression.groovy]: BIGTOP-218 is really very old(created on 2011). At that time snappy may not be included. And today bigtop already has libsnappy built-in, so this case can be enabled. And test passed on both x86 and aarch64.

Comment (evans-ye) [inline bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestHBaseCompression.groovy]: Then how about remove the line?

File: bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestImportTsv.groovy (hunk 1 above)
Comment (evans-ye) [inline bigtop-tests/test-artifacts/hbase/src/main/groovy/org/apache/bigtop/itest/hbase/smoke/TestImportTsv.groovy]: No prob. I see other JIRAs addressing the DEPRECATED issue of rmr, hence I think it's a good time to address it here as well.

File: bigtop-tests/smoke-tests/hbase/build.gradle [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+def tests_to_include() {
+  return [
```
Comment (JunHe77) [inline bigtop-tests/smoke-tests/hbase/build.gradle]: which line? 18 or 19? I found that line 19 is 2 spaces indentaion.

Comment (c0s) [inline bigtop-tests/smoke-tests/hbase/build.gradle]: The content of the returned list: lines 20 to 30

Comment (JunHe77) [inline bigtop-tests/smoke-tests/hbase/build.gradle]: Got it. Thanks!

**Comment by c0s** [comment]
+1 assuming the tests are working now (I haven't tested it myself though)

**Comment by c0s** [comment]
Hello. Any comments on the passing of the tests with this new version of the patch? Thanks!

**Comment by JunHe77** [comment]
Hi, c0s,
This smoke test is tested on ubuntu-16.04/amd64.

**Comment by c0s** [comment]
I have pushed the fix in, thanks! Please close this PR manually - as I have skipped the magic words in the commit message ;)

**Comment by JunHe77** [comment]
Thanks for merging, c0s.

**Pull request closed without merging**